plutus-playground-server/test/Playground/UsecasesSpec.hs

```diff
@@ -320,98 +363,124 @@ crowdfundingTest :: TestTree
-              , mkSimulatorWallet w2 (Ada.lovelaceValueOf 19999990)
-              , mkSimulatorWallet w3 (Ada.lovelaceValueOf 19999990)
+              [ mkSimulatorWallet w1 $ Ada.lovelaceValueOf 60
+              , mkSimulatorWallet w2 $ Ada.lovelaceValueOf 19
+              , mkSimulatorWallet w3 $ Ada.lovelaceValueOf 20
+              , mkSimulatorWallet w4 $ Ada.lovelaceValueOf 21
               ]
         , testCase "should run failed campaign and return the funds" $
           evaluate failedCampaign >>=
           hasFundsDistribution
-              [ mkSimulatorWallet w1 twentyAda
-              , mkSimulatorWallet w2 twentyAda
-              , mkSimulatorWallet w3 twentyAda
+              [ mkSimulatorWallet w1 $ Ada.lovelaceValueOf 20
+              , mkSimulatorWallet w2 $ Ada.lovelaceValueOf 20
+              , mkSimulatorWallet w3 $ Ada.lovelaceValueOf 20
               ]
         ]
   where
-    twentyAda = Ada.adaValueOf 20
+    twentyLovelace = Ada.lovelaceValueOf 20
     successfulCampaign =
         Evaluation
             { wallets =
-                  [ mkSimulatorWallet w1 twentyAda
-                  , mkSimulatorWallet w2 twentyAda
-                  , mkSimulatorWallet w3 twentyAda
+                  [ mkSimulatorWallet w1 $ Ada.lovelaceValueOf 30
+                  , mkSimulatorWallet w2 $ Ada.lovelaceValueOf 30
+                  , mkSimulatorWallet w3 $ Ada.lovelaceValueOf 30
+                  , mkSimulatorWallet w4 $ Ada.lovelaceValueOf 30
                   ]
             , program =
                   toJSONString
-                      [ AddBlocks 1
+                      [ CallEndpoint
+                            { caller = w1
+                            , argumentValues =
+                                  FunctionSchema
+                                      { endpointName =
+                                            EndpointName "schedule collection"
+                                      , arguments = toEndpointParam ()
+                                      }
+                            }
                       , CallEndpoint
-                            { endpointName = EndpointName "schedule collection"
-                            , wallet = w1
-                            , arguments = toEndpointParam ()
+                            { caller = w2
+                            , argumentValues =
+                                  FunctionSchema
+                                      { endpointName = EndpointName "contribute"
+                                      , arguments =
+                                            toEndpointParam
+                                                Contribution
+                                                    { contribValue =
+                                                          Ada.lovelaceValueOf 11
+                                                    }
+                                      }
```

Comment:
`where contributeScheme :: Ada -> FunctionScheme`?

Reply:
Yes. Refactored the whole file along these lines. Much nicer. Thanks.